x-pack/plugins/stack_connectors/public/connector_types/cases_webhook/steps/get.tsx

```diff
@@ -6,103 +6,187 @@
+  });
+  const { getIncidentMethod = WebhookMethods.GET } = config ?? {};
+
+  return (
+    <span data-test-subj="getStep" style={{ display: display ? 'block' : 'none' }}>
+      <EuiText>
+        <h3>{i18n.STEP_3}</h3>
+        <small>
+          <p>{i18n.STEP_3_DESCRIPTION}</p>
+        </small>
+      </EuiText>
+      <EuiFlexGroup direction="column">
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <UseField
+              path="config.getIncidentMethod"
+              component={Field}
+              config={{
+                label: i18n.GET_INCIDENT_METHOD,
+                defaultValue: 'get',
+                type: FIELD_TYPES.SELECT,
+                validations: [
+                  {
+                    validator: emptyField(i18n.GET_METHOD_REQUIRED),
+                  },
+                ],
+              }}
+              css={css`
+                .euiFormRow__labelWrapper {
+                  margin-bottom: 9px;
+                }
+              `}
+              componentProps={{
+                euiFieldProps: {
+                  'data-test-subj': 'webhookGetIncidentMethodSelect',
+                  options: ['get', 'post'].map((verb) => ({
+                    text: verb.toUpperCase(),
+                    value: verb,
+                  })),
+                  readOnly,
+                },
+              }}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <UseField
+              path="config.getIncidentUrl"
+              config={{
+                label: i18n.GET_INCIDENT_URL,
+                validations: [
+                  {
+                    validator: urlField(i18n.GET_INCIDENT_URL_REQUIRED),
+                  },
+                  { validator: containsExternalIdForGet(getIncidentMethod) },
+                ],
+                helpText: i18n.GET_INCIDENT_URL_HELP,
+              }}
+              component={MustacheTextFieldWrapper}
+              componentProps={{
+                euiFieldProps: {
+                  readOnly,
+                  'data-test-subj': 'webhookGetUrlText',
+                  messageVariables: urlVars,
+                  paramsProperty: 'getIncidentUrl',
+                  buttonTitle: i18n.ADD_CASES_VARIABLE,
+                  showButtonTitle: true,
+                },
+              }}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        {getIncidentMethod === WebhookMethods.POST ? (
+          <EuiFlexItem>
+            <UseField
+              path="config.getIncidentJson"
+              config={{
+                helpText: i18n.GET_INCIDENT_JSON_HELP,
+                label: i18n.GET_INCIDENT_JSON,
+                validations: [
+                  {
+                    validator: requiredJsonForPost(getIncidentMethod),
+                  },
+                  {
+                    validator: containsExternalIdForPost(getIncidentMethod),
```

Comment:
I'm not sure we need this here. If the ID can still go in the URL(for whatever reason, these APIs are weird) we might be restricting some users to always send the ID in the body when they don't need to.

Reply:
I agree. I would say to not validate the POST body or URL as we are not sure where and if the ID should go.

Reply:
[removed](https://github.com/elastic/kibana/pull/197437/commits/5ab48360718d8dda6fa858bc2b9ca0e215ae246b)